src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -63,6 +83,13 @@ private enum Flags : byte
         private SynthesizedSealedPropertyAccessor _lazySynthesizedSealedAccessor;
         private CustomAttributesBag<CSharpAttributeData> _lazyCustomAttributesBag;
 
+        /// <summary>
+        /// Represents that _lazyBackingFieldSymbolForFieldKeyword value is unknown.
+        /// </summary>
+        private static readonly object _lazyBackingFieldSymbolForFieldKeywordSentinel = new object();
+
+        private object _lazyBackingFieldSymbolForFieldKeyword = _lazyBackingFieldSymbolForFieldKeywordSentinel;
```

Comment:
>_lazyBackingFieldSymbolForFieldKeyword [](http://example.com/codeflow?start=23&length=38)

I think this should be renamed back to ```_lazyBackingFieldSymbol``` the property symbol should store only one symbol for the backing field, regardless of the reason why it was created.

Reply:
@AlekseyTs Do you mean we will use it for both `BackingField` and `BackingFieldForFieldKeyword`?

This can cause the value of `BackingField` to change during the object life time, depending on whether we have called `BackingFieldForFieldKeyword` before or not.

Reply:
> Do you mean we will use it for both BackingField and BackingFieldForFieldKeyword?

Yes. Their implementation should be adjusted accordingly in order to preserve original behavior.

> This can cause the value of ```BackingField``` to change during the object life time, depending on whether we have called ```BackingFieldForFieldKeyword``` before or not.

Not if we implement it correctly. Please read my comment for ```_lazyBackingFieldSymbolForFieldKeyword``` in commit 49 again. Especially this part: "We might need to record an extra information about whether the symbol was created by constructor. That can be recorder in the ```SynthesizedBackingFieldSymbol``` the same way we record whether it was created for a ```field``` keyword."